src/containers/CalculatorContainer.js

```diff
@@ -1,9 +1,13 @@
 import { connect } from 'react-redux'
+
 import Calculator from '../components/Calculator/Calculator'
+import subjectPicker from '../store/selectors/subjectPicker'
+import { PRIMARY_SUBJECTS, LINGUAL_SUBJECTS } from '../constants/SubjectTypes'
 
-function mapStateToProps (state) {
-  return { subjects: state.subjects }
-}
+const mapStateToProps = (state) => ({
+  maxPrimaryScoreWrapper: subjectPicker(state.subjects[PRIMARY_SUBJECTS], PRIMARY_SUBJECTS),
```

Comment:
subjectPicker filters subjects within itself by provided subjectType, so you can just provide `state.subjects` as the first argument.